src/metabase/integrations/common.clj

```diff
@@ -28,14 +28,22 @@
     ;; remove membership from any groups as needed
     (when (seq to-remove)
       (log/debugf "Removing user %s from group(s) %s" user-id to-remove)
-      (db/delete! PermissionsGroupMembership :group_id [:in to-remove], :user_id user-id))
+      (try
+       (db/delete! PermissionsGroupMembership :group_id [:in to-remove], :user_id user-id)
+       (catch clojure.lang.ExceptionInfo e
+         ;; in case sync attempts to delete the last admin, the pre-delete hooks of
+         ;; [[metabase.models.permissions-group-membership/PermissionsGroupMembership]] will throw an exception.
+         ;; but we don't want to block user from logging-in, so catch this exception and log a warning
+         (if (= (ex-message e) (str pgm/fail-to-remove-last-admin-msg))
+           (log/warn "Attempted to remove the last admin.")
```

Comment:
I think we could make this error message a bit more actionable for admins.

Here's my suggestion:
`
Attempted to remove the last admin during group sync! Check your SSO group mappings and make sure the Administrators group is mapped correctly.
`